lib/pengine/pe_actions.c

```diff
@@ -1632,21 +1631,13 @@ pe__is_newer_op(const xmlNode *xml_a, const xmlNode *xml_b,
 
     const char *a_node = crm_element_value(xml_a, PCMK__META_ON_NODE);
     const char *b_node = crm_element_value(xml_b, PCMK__META_ON_NODE);
-    bool same_node = true;
-
-    /* @COMPAT The on_node attribute was added to last_failure as of 1.1.13 (via
-     * 8b3ca1c) and the other entries as of 1.1.12 (via 0b07b5c).
-     *
-     * In case that any of the PCMK__XE_LRM_RSC_OP entries doesn't have on_node
-     * attribute, we need to explicitly tell whether the two operations are on
-     * the same node.
-     */
-    if (a_node == NULL || b_node == NULL) {
-        same_node = same_node_default;
+    bool same_node = false;
 
-    } else {
-        same_node = pcmk__str_eq(a_node, b_node, pcmk__str_casei);
-    }
+    CRM_CHECK((a_xml_id != NULL) && (b_xml_id != NULL)
```

Comment:
I mentioned this in the commit message, but I'm second-guessing the `CRM_CHECK()` here on malformed input XML. Granted, it would have to be malformed within a CIB status section, which is something only Pacemaker should ever touch.

Reply:
drop the check, the string comparison is sufficient